### PR TITLE
refactor: telescope adapter `search_and_add`

### LINKED ISSUE #14: `CdProjectManualAdd` hangs when using on Mac
This is because it's also searching through the hidden `Library/` directory. 

I'll try taking charge on addressing this since i wrote the feat, I think a rewrite of this function would be best. It would also address [this comment](https://github.com/LintaoAmons/cd-project.nvim/blob/9a0b83e1bedf03cae5f0087ed2e5b3f523f50c77/lua/cd-project/adapter/telescope.lua#L64)

### PR DESCRIPTION
**Pull request by vague2k**
Hi!

This PR tries to make the telescope adapter's implementation of `manual_cd_project` now named `search_and_add`. More robust.

## What does this PR do?
- refactor `utils.check_for_find_cmd` in such a way that when we pass the command string to [this function](https://github.com/LintaoAmons/cd-project.nvim/blob/2e16beac269a2401d9be383fdc2f02f2ec7c4600/lua/cd-project/adapter/telescope.lua#L68), i.e `vim.fn.jobstart(find_command...)`, we already have the directories we want to search for in `stdout_buffered` instead of using `cwd`.

- `utils.check_for_find_command` does not take into account the `find` command, as it would cause formatting issues and *in my humble opinion* make for a complex and hard to follow implementation. 

- deprecate `utils.format_based_on_dir` since fd will handle path string formatting regardless of OS.

- adds a new nvim user command being `CdSearchAndAdd`

edit: I also feel it's worth noting that the `utils.check_for_find_cmd` refactor also strives to resolve #14

### CODE REVIEW

**Comment by LintaoAmons** [comment]
Didn't go through thoroughly, but I think you did a good job~ 

One thing I noticed is that maybe you are dealing with filepath in different os. Maybe can check `vim.fs.normalise`

**Comment by LintaoAmons** [comment]
Sorry, wrong button

**Comment by vague2k** [comment]
I've been writing my own plugin that uses `vim.fs.normalize`. I'll look into using this for this plugin. 

Remember that `vim.fs.normalize` would make this plugin only work in NeoVim => 0.8.0
